### PR TITLE
Add verbose mode in kw_db to show sqlite3 database commands

### PR DESCRIPTION
**Pull request by aquilamacedo**
In certain feature functions, there's a reliance on `kw_db` functions. This commit introduces support for a verbose mode, revealing the SQL commands executed under the hood

### CODE REVIEW

File: src/kernel_config_manager.sh [hunk 1]
```diff
@@ -126,8 +126,10 @@ function save_config_file()
     return 2 # ENOENT
   fi
 
+  cmd="$(select_from "${flag}" "kernel_config WHERE name IS '${config_name}'")"
+  show_verbose "$flag" "$cmd"
```
Comment (rodrigosiqueira) [inline src/kernel_config_manager.sh:130]: I think you need to reverse this order and use `SILENT` in the `select_from`.

File: src/kernel_config_manager.sh [hunk 2]
```diff
@@ -126,8 +126,10 @@ function save_config_file()
...
   # Checks if there is already an entry for that kernel config file in the database
-  is_on_database="$(select_from "kernel_config WHERE name IS '${config_name}'")"
+  is_on_database="$(select_from 'SILENT' "kernel_config WHERE name IS '${config_name}'")"
```
Comment (rodrigosiqueira) [inline src/kernel_config_manager.sh:132]: I think you forgot the `show_verbose`.

File: src/kernel_config_manager.sh [hunk 3]
```diff
@@ -516,7 +518,7 @@ function list_configs()
 {
   local configs
 
-  configs="$(select_from 'kernel_config' 'name AS "Name", description AS "Description", last_updated_datetime AS "Last updated"' '.mode column')"
+  configs="$(select_from 'SILENT' 'kernel_config' 'name AS \"Name\", description AS \"Description\", last_updated_datetime AS \"Last updated\"' '.mode column')"
```
Comment (rodrigosiqueira) [inline src/kernel_config_manager.sh:521]: You forgot to add the `show_verbose` before this function.

File: src/kernel_config_manager.sh [hunk 4]
```diff
@@ -549,13 +551,17 @@ function basic_config_validations()
...
+  local cmd
 
   if [[ ! -f "${kernel_configs_dir}/${config_name}" ]]; then
     complain "Couldn't find config file named: ${config_name}"
     exit 2 # ENOENT
   fi
 
-  query_output=$(select_from "kernel_config WHERE name IS '${config_name}'")
+  cmd="$(select_from "${flag}" "kernel_config WHERE name IS '${config_name}'")"
+  show_verbose "$flag" "$cmd"
```
Comment (rodrigosiqueira) [inline src/kernel_config_manager.sh:562]: I think you need to reverse this order and use `SILENT` in the `select_from`.

File: src/pomodoro.sh [hunk 5]
```diff
@@ -79,16 +79,20 @@ function show_active_pomodoro_timebox()
...
 # Show registered tags with number identification.
 function show_tags()
 {
-  local flag="$1"
+  local flag=${1:-'SILENT'}
   local tags
+  local cmd
+
+  cmd=$(select_from "$flag" 'tag WHERE "active" IS 1' '"id" AS "ID", "name" AS "Name"' '.mode column' 'id')
+  show_verbose "$flag" "$cmd"
```
Comment (rodrigosiqueira) [inline src/pomodoro.sh:93]: I think you need to reverse this order and use `SILENT` in the `select_from`.

File: src/pomodoro.sh [hunk 6]
```diff
@@ -124,8 +128,12 @@ function is_tag_already_registered()
   local flag="$1"
   local tag_name="$2"
   local is_tag_registered=''
+  local cmd
+
+  cmd=$(select_from "$flag" "tag WHERE name IS '${tag_name}'")
+  show_verbose "$flag" "$cmd"
```
Comment (rodrigosiqueira) [inline src/pomodoro.sh:134]: I think you need to reverse this order and use `SILENT` in the `select_from`.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
I added some comments to this PR; please review everything based on that.

**Comment by rodrigosiqueira** [comment]
Hi @aquilamacedo , please, update this PR. This is the final part to complete the verbose task.

**Comment by aquilamacedo** [comment]
Hi @rodrigosiqueira. Yes, I'm working on it, I'm trying to think of a logic to create a new functionality within `kwlib.sh`, something like a VERBOSE_DB (as in the screenshot below), it would be a good approach to show the verbose output of the `kw_db.sh` functions.

image

However, I'm still uncertain about the best approach to initialize the 'VERBOSE_DB' value (for instance, within the 'kernel_config_manager'). This is because there already exists 'VERBOSE', which can be initialized if the --verbose argument is passed. Therefore, the flag will assume the value initialized by this line:

```
   [[ -n "${options_values['VERBOSE']}" ]] && flag='VERBOSE'
```

Integrating 'VERBOSE_DB' requires a similar strategy to ensure proper initialization that aligns with the existing code logic.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
@davidbtadokoro since you discussed with @aquilamacedo , could you review this change?

**Comment by davidbtadokoro** [comment]
> @davidbtadokoro since you discussed with @aquilamacedo , could you review this change?

For sure! Sorry for the delay, starting to work on it now

File: src/pomodoro.sh [hunk 7]
```diff
@@ -85,10 +85,13 @@ function show_active_pomodoro_timebox()
 # Show registered tags with number identification.
```
Comment (davidbtadokoro) [inline src/pomodoro.sh:85]: Think you forgot to add verbose to `select_from` in the function `show_active_pomodoro_timebox` that `kw pomodoro --check-timer` uses underneath.

It is
Screenshot from 2024-01-17 14-06-43


but it should be
Screenshot from 2024-01-17 14-09-03

Comment (davidbtadokoro) [inline src/pomodoro.sh:85]: Also, think you forgot to add verbose to `select_from` in the function `get_tag_name`.

**Review by davidbtadokoro — CHANGES_REQUESTED** [review]
Hey @aquilamacedo, sorry for the delay and thanks so much for this change. We are almost at the perfect point :+1:

**Review by davidbtadokoro — CHANGES_REQUESTED** [review]
Hi @aquilamacedo, and thanks for updating the PR again! Left fewer and simpler comments this time :grin:

We really are getting into that perfect state, so much so that, with the greenlight of @rodrigosiqueira, I can address these comments myself and merge it into the unstable branch.

File: tests/unit/lib/kwlib_test.sh [hunk 8]
```diff
@@ -1003,4 +1003,30 @@ function test_check_if_create_shared_memory_dir_creates_dir_in_default_mktemp_di
   assertTrue "${LINENO}: Did not create directory at /tmp/ dir" '[[ $output =~ /tmp/ ]]'
 }
 
+function test_cmd_manager_check_silent_option()
```
[the remark this answers is not in the document]
Comment (aquilamacedo) [reply tests/unit/lib/kwlib_test.sh]: I liked this test @davidbtadokoro, I'm going to use it :)

File: src/lib/kwlib.sh [hunk 9]
```diff
@@ -111,6 +111,9 @@ function cmd_manager()
       say "$command_for_eval"
       return 0
       ;;
+    VERBOSE_DB)
```
[the remark this answers is not in the document]
Comment (aquilamacedo) [reply src/lib/kwlib.sh]: I'm cool with the name being lengthy. It's actually good because it gives a clear picture of what it's all about.

**Review by davidbtadokoro — COMMENTED** [review]
Thanks again @aquilamacedo. IMHO, I can't see other improvments to this PR. @rodrigosiqueira, I'll wait for your greenlight.

**Comment by rodrigosiqueira** [comment]
Hi, thanks a lot for keeping improving this PR!
Change merged into the unstable branch.

**Pull request closed without merging**